asmcomp/cmmgen.ml

```diff
@@ -1773,6 +1775,39 @@ and transl_prim_1 env p arg dbg =
   | Pint_as_pointer ->
      Cop(Caddi, [transl env arg; Cconst_int (-1)])
      (* always a pointer outside the heap *)
+  (* Pointer operations *)
+  | Pload8 ->
+     tag_int (Cop(Cload Byte_unsigned,
+                  [transl_unbox_int env Pnativeint arg]))
+  | Pload16 aligned ->
+     let unboxed_arg = transl_unbox_int env Pnativeint arg in
+     tag_int (
+       if aligned then
+         Cop(Cload Sixteen_unsigned, [unboxed_arg])
+       else unaligned_load_16 unboxed_arg (Cconst_int 0))
+  | Pload(Pint32, aligned) ->
+     let unboxed_arg = transl_unbox_int env Pnativeint arg in
+     box_int dbg Pint32 (
+       if aligned then
+         Cop(Cload Thirtytwo_unsigned, [unboxed_arg])
+       else unaligned_load_32 unboxed_arg (Cconst_int 0))
+  | Pload(Pint64, aligned) ->
+     assert(size_int = 8);
+     let unboxed_arg = transl_unbox_int env Pnativeint arg in
+     box_int dbg Pint64 (
+       if aligned then
+         Cop(Cload Word_val, [unboxed_arg])
+       else unaligned_load_64 unboxed_arg (Cconst_int 0))
+  | Pload(Pnativeint, aligned) ->
```

Comment:
We don't have primitves for loading/storing nativeints for strings/bigstrings so we don't need ones for raw pointers. One can implement them in terms of the int32/int64 primitives and it should be as efficient.